content/en/blog/_posts/2025-08-27-Kubernetes-v1-34-Release/index.md

```diff
@@ -0,0 +1,512 @@
+
+### Stable: Automated cgroup driver detection
+
+Historically, configuring the correct cgroup driver has been a pain point for users running Kubernetes clusters. This feature instructs the `kubelet` to ask the CRI implementation which cgroup driver to use. It was introduced in v1.28 and has graduated to stable in v1.34.  
+The `kubelet` command line flag `--cgroup-driver` (and its alternative `cgroupDriver` in KubeletConfiguration) is marked as deprecated and in two releases or later it will be removed. This will make it impossible to use Kubernetes with the containerd 1.X versions. The timeline of this deprecation is aligned with containerd 1.7 EOL. This is an early warning that if you are using containerd 1.X, consider switching to 2.0+ soon.
+
+This work was done as part of [KEP \#4033](https://kep.k8s.io/4033) led by SIG Node.
+
+### Beta: Projected ServiceAccount tokens for `kubelet` image credential providers
+
+The `kubelet` credential providers, used for pulling private container images, traditionally relied on long-lived secrets stored on the node or in the cluster. This approach increased security risks and management overhead, as these credentials were not tied to the specific workload and did not rotate automatically.  
+To solve this, the `kubelet` can now request short-lived, audience-bound ServiceAccount tokens for authenticating to container registries. This allows image pulls to be authorized based on the Pod's own identity rather than a node-level credential.  
+The primary benefit is a significant security improvement. It eliminates the need for long-lived secrets for image pulls, reducing the attack surface and simplifying credential management for both administrators and developers
+
+This work was done as part of [KEP \#4412](https://kep.k8s.io/4412) led by SIG Auth.
+
+### Alpha: Support for KYAML, a Kubernetes dialect of YAML
+
+KYAML aims to be a safer and less ambiguous YAML subset, and was designed specifically for Kubernetes. Whatever version of Kubernetes you use, starting from Kubernetes v1.34 you are able to use KYAML for writing manifests and/or Helm charts.
+   
+KYAML addresses specific challenges with both YAML and JSON. YAML's significant whitespace requires careful attention to indentation and nesting, while its optional string-quoting can lead to unexpected type coercion (for example: ["The Norway Bug"](https://hitchdev.com/strictyaml/why/implicit-typing-removed/)). Meanwhile, JSON lacks comment support and has strict requirements for trailing commas and quoted keys. 
+  
+You can write KYAML and pass it as an input to any version of `kubectl`, because all KYAML files are also valid as YAML. With `kubectl` v1.34, you are also able to request KYAML output (as in `kubectl get -o kyaml …`). If you prefer, you can still request the output in JSON or YAML format.
+
+This work was done as part of [KEP \#5295](https://kep.k8s.io/5295) led by SIG CLI.
+
+### \<Stable/Beta/Alpha\>: \<THEME 4\>
```

Comment:
one thing in 1.34 that may be breaking customer is the static pod are not allowed to have API objects: https://github.com/kubernetes/kubernetes/pull/131837

This is a change of behavior that may require to change static pod definition and deserves to be highlighted.

cc: @BenTheElder

Reply:
I'm not sure we have a place for changes like this in the Release Announcement 🤔 I would make sure this is covered properly in the release notes.